web/src/emoji.ts

```diff
@@ -121,19 +181,24 @@ function build_emojis_by_name({
     emoji_catalog,
     get_emoji_name,
     default_emoji_aliases,
-}) {
+}: {
+    realm_emojis: typeof active_realm_emojis;
+    emoji_catalog: EmojiCodes["emoji_catalog"];
+    get_emoji_name: (codepoint: string) => string | undefined;
+    default_emoji_aliases: Map<string, string[]>;
+}): Map<string, EmojiDict> {
     // Please keep this as a pure function so that we can
     // eventually share this code with the mobile codebase.
-    const map = new Map();
+    const map = new Map<string, EmojiDict>();
 
     for (const codepoints of Object.values(emoji_catalog)) {
         for (const codepoint of codepoints) {
             const emoji_name = get_emoji_name(codepoint);
             if (emoji_name !== undefined) {
-                const emoji_dict = {
+                const emoji_dict: EmojiDict = {
                     name: emoji_name,
                     display_name: emoji_name,
-                    aliases: default_emoji_aliases.get(codepoint),
+                    aliases: default_emoji_aliases.get(codepoint) || [],
                     is_realm_emoji: false,
                     emoji_code: codepoint,
                     has_reacted: false,
```

Comment:
Should the next loop also use `const emoji_dict: EmojiDict`? It's a different definition, so it seems helpful to write down that we intend this to be that type.